examples/text/plot_hashing_vs_dict_vectorizer.py

```diff
@@ -3,109 +3,380 @@
+# “a”, “is” in English) which do not carry meaningful information about the
+# actual contents of a document. If we were to feed the word count data directly
+# to a classifier those very common terms would shadow the frequencies of rarer
+# yet more interesting terms. In order to re-weight the count features into
+# floating point values suitable for usage by a classifier it is very common to
+# use the tf–idf transform as implemented by the
+# :func:`~sklearn.feature_extraction.text.TfidfTransformer`. TF stands for
+# "term-frequency" while "tf–idf" means term-frequency times inverse
+# document-frequency.
+#
+# We now benchmark the :func:`~sklearn.feature_extraction.text.TfidfVectorizer`,
+# which is equivalent to combining the tokenization and occurrence counting of
+# the :func:`~sklearn.feature_extraction.text.CountVectorizer` along with the
+# normalizing and weighting from a
+# :func:`~sklearn.feature_extraction.text.TfidfTransformer`.
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+t0 = time()
+vectorizer = TfidfVectorizer()
+vectorizer.fit_transform(raw_data)
+duration = time() - t0
+dict_count_vectorizers["vectorizer"].append(vectorizer.__class__.__name__)
+dict_count_vectorizers["speed"].append(data_size_mb / duration)
+print(f"done in {duration:.3f}s at {data_size_mb / duration:.3f}MB/s")
+print(f"Found {len(vectorizer.get_feature_names_out())} unique terms")
+
+# %%
+# Let's conclude this notebook by summarizing all the recorded processing speeds
+# in a single plot:
+
+fig, ax = plt.subplots(figsize=(12, 6))
+
+y_pos = np.arange(len(dict_count_vectorizers["vectorizer"]))
+ax.barh(y_pos, dict_count_vectorizers["speed"], align="center")
+ax.set_yticks(y_pos)
+ax.set_yticklabels(dict_count_vectorizers["vectorizer"])
+ax.invert_yaxis()
+_ = ax.set_xlabel("speed (MB/s)")
+
+# %%
+# Notice from the plot that
+# :func:`~sklearn.feature_extraction.text.TfidfVectorizer` is slightly slower
+# than :func:`~sklearn.feature_extraction.text.CountVectorizer` because of the
+# extra operation induced by the
+# :func:`~sklearn.feature_extraction.text.TfidfTransformer`. Also notice that,
+# by setting the number of features `n_features = 2**18`, the
+# :func:`~sklearn.feature_extraction.text.HashingVectorizer` performs much
+# faster than the :func:`~sklearn.feature_extraction.text.CountVectorizer`. Keep
+# in mind that this may cause hash collisions and no inverse transform can be
+# used to recover the original dictionary.
+#
+# We highlight that :func:`~sklearn.feature_extraction.text.CountVectorizer` and
+# :func:`~sklearn.feature_extraction.text.HashingVectorizer` are more efficient
+# than using their equivalent :func:`~sklearn.feature_extraction.DictVectorizer`
+# and :func:`~sklearn.feature_extraction.FeatureHasher` on manually tokenized
+# documents. The reason is that the internal tokenization step of the formers
+# compiles a regular expression once and then reuses it for all the documents.
```

Comment:
```suggestion
# We highlight that :func:`~sklearn.feature_extraction.text.CountVectorizer` and
# :func:`~sklearn.feature_extraction.text.HashingVectorizer` perform better 
# than their equivalent :func:`~sklearn.feature_extraction.DictVectorizer`
# and :func:`~sklearn.feature_extraction.FeatureHasher` on manually tokenized
# documents since the internal tokenization step of the former vectorizers
# compiles a regular expression once and then reuses it for all the documents.
```